### PR TITLE
Adds the ability to sort table data

### PR DESCRIPTION
**Pull request by anmoljagetia**
@GordonSmith @mlzummo Please review!
Adds the ability to sort the data in the table based on column selected. 
Demo at : http://rawgit.com/anmoljagetia/Visualization/tableSort/demos/dermatology.html?src/other/Table

### CODE REVIEW

File: src/other/Table.css [hunk 1]
```diff
@@ -6,13 +6,19 @@
...
 	background-color:#1f77b4;
 	border-width: 1px;
 	border-style: solid;
 	border-color: #a9c6c9;
     color:white;
     white-space: nowrap;
+    cursor : pointer;
+}
+
+
```
Comment (GordonSmith) [inline src/other/Table.css]: Remove additional blank lines

File: src/other/Table.css [hunk 2]
```diff
@@ -6,13 +6,19 @@
...
+.thIcon {
```
Comment (GordonSmith) [inline src/other/Table.css]: Should be namespaced with .other_Table

File: src/other/Table.js [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+}(this, function (d3, HTMLWidget, FAChar, Paginator) {
     function Table() {
         HTMLWidget.call(this);
         this._tag = "div";
-
+        this._order = -1;
+        this._currentSort = "";
         this._columns = [];
         this._paginator = new Paginator();
+        this._faChar = new FAChar();
```
Comment (GordonSmith) [inline src/other/Table.js]: faChar not used.

File: src/other/Table.js [hunk 4]
```diff
@@ -55,10 +57,45 @@
...
+            .on("click", function (column) {
+                if (context._currentSort === column) {
+                    if(context._order !== -1) {
+                        context._order = 1;
+                    }
+                } else {
+                    context._order = -1;
+                }
+                context.click(column)
+                ;
```
Comment (GordonSmith) [inline src/other/Table.js]: move semicolon up a line

File: src/other/Table.js [hunk 5]
```diff
@@ -55,10 +57,45 @@
...
+        ;
+        th.select(".thText")
+            .text(function (column) {
+                return column;
+            })
         ;
+        th.select(".thIcon")
+            .text(function (column) {
+                if (context._order === -1) {
+                    // return context._currentSort === column ? "\uf106" : "";
```
Comment (GordonSmith) [inline src/other/Table.js]: remove commented code.

File: src/other/Table.js [hunk 6]
```diff
@@ -138,8 +175,31 @@
     };
     
 
-    Table.prototype.click = function (d) {
-    };
+    Table.prototype.click = function (column) {
```
Comment (GordonSmith) [inline src/other/Table.js]: This "click" method is already in use by the data row.  Can you rename it to "_headerClick" or such like (the "_" is to indicate its private).

File: src/other/Table.css [hunk 7]
```diff
@@ -6,13 +6,18 @@
 }
 
 .other_Table th {
-    padding:5px 10px;
+	padding:5px 10px;
 	background-color:#1f77b4;
 	border-width: 1px;
 	border-style: solid;
 	border-color: #a9c6c9;
```
Comment (GordonSmith) [inline src/other/Table.css]: indentation is off...

File: src/other/Table.js [hunk 8]
```diff
@@ -1,15 +1,16 @@
...
 (function (root, factory) {
     if (typeof define === "function" && define.amd) {
-        define(["../common/HTMLWidget","../other/Paginator", "css!./Table"], factory);
+        define(["d3", "../common/HTMLWidget","../common/FAChar","../other/Paginator","css!./Table"], factory);
     } else {
-        root.other_Table = factory(root.common_HTMLWidget, root.other_Paginator);
+        root.other_Table = factory(root.d3, root.common_HTMLWidget, root.common_FAChar, root.other_Paginator);
     }
-}(this, function (HTMLWidget, Paginator) {
+}(this, function (d3, HTMLWidget, FAChar, Paginator) {
```
Comment (GordonSmith) [inline src/other/Table.js]: FAChar not used

**Comment by GordonSmith** [comment]
Just the minor indentation and the unused FAChar, then I can pull.

File: src/other/Table.css (hunk 7 above)
Comment (anmoljagetia) [inline src/other/Table.css]: My text editor (sublime text), doesn't seem to show anything wrong, nor does vim :/
selection_005
selection_006

Comment (mlzummo) [inline src/other/Table.css]: there is setting for sublime to remove all whitespace on save (that helps a lot)

Comment (anmoljagetia) [inline src/other/Table.css]: @mlzummo thanks :) I think it fixes the issue! Don't really know what causes it.

**Comment by GordonSmith** [comment]
There should be a setting to convert tabs to spaces (1 tab === 4 spaces), which will fix the indentation.

I think there is a lint setting for that, I should set!

File: src/other/Table.css (hunk 7 above)
Comment (GordonSmith) [inline src/other/Table.css]: I _think_ that setting just removes trailing white spaces?

File: src/other/Table.js [hunk 9]
```diff
@@ -55,10 +56,42 @@
...
+        ;
+        th.select(".thText")
+            .text(function (column) {
+                return column;
+            })
         ;
+        th.select(".thIcon")
+            .text(function (column) {
+                if (context._order === -1) {
+                    return context._currentSort === column ? "\t" + "\uf078" : "";
```
Comment (GordonSmith) [inline src/other/Table.js]: I don't think HTML support tab spaces "\t" - if you want a "gap" you should either style the span or add a "&nbsp;"

**Comment by GordonSmith** [comment]
Looks good.

**Pull request merged**